homeassistant/components/abode/__init__.py

```diff
@@ -110,69 +106,72 @@ def __init__(self, username, password, cache, name, polling, exclude, lights):
         )
         self.name = name
         self.polling = polling
-        self.exclude = exclude
-        self.lights = lights
         self.devices = []
 
-    def is_excluded(self, device):
-        """Check if a device is configured to be excluded."""
-        return device.device_id in self.exclude
 
-    def is_automation_excluded(self, automation):
-        """Check if an automation is configured to be excluded."""
-        return automation.automation_id in self.exclude
+async def async_setup(hass, config):
+    """Set up Abode component."""
+    if DOMAIN not in config:
+        return True
 
-    def is_light(self, device):
-        """Check if a switch device is configured as a light."""
-        import abodepy.helpers.constants as CONST
+    conf = config[DOMAIN]
 
-        return device.generic_type == CONST.TYPE_LIGHT or (
-            device.generic_type == CONST.TYPE_SWITCH and device.device_id in self.lights
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=deepcopy(conf)
         )
+    )
 
+    return True
 
-def setup(hass, config):
-    """Set up Abode component."""
-    from abodepy.exceptions import AbodeException
 
-    conf = config[DOMAIN]
-    username = conf.get(CONF_USERNAME)
-    password = conf.get(CONF_PASSWORD)
-    name = conf.get(CONF_NAME)
-    polling = conf.get(CONF_POLLING)
-    exclude = conf.get(CONF_EXCLUDE)
-    lights = conf.get(CONF_LIGHTS)
+async def async_setup_entry(hass, config_entry):
+    """Set up Abode component via config entry (Integrations UI)."""
+    username = config_entry.data.get(CONF_USERNAME)
+    password = config_entry.data.get(CONF_PASSWORD)
+    name = config_entry.data.get(CONF_NAME)
```

Comment:
We aren't saving the extra options available in the config yaml in the config entry data. We need to do that if they should be available here.

Reply:
Where do we use the name option?

Reply:
I’m not quite sure actually. A lot of this is obviously from whoever created this component previously. I just assumed the name option was an override for a property that Home Assistant used. I personally don’t see a need for it and think it’s simpler using the default which is the component name, especially since we can only have one instance right now.